core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -335,14 +348,16 @@ Integer convert(JsonNode element) {
     @Override
     void validate(JsonNode element) {
       Preconditions.checkArgument(
-          element.isInt(), "Cannot parse integer from non-int value: %s", element);
+          element.isInt(), "Cannot parse integer from non-int value: %s: %s", property, element);
```

Comment:
Is this correct? This is an element in the array, not the array property itself. How about using `value in %s: %s` to be more clear?

Reply:
makes sense, done